src/api/routes/ft.ts

```diff
@@ -121,11 +122,18 @@ const ShowRoutes: FastifyPluginCallback<Record<never, never>, Server, TypeBoxTyp
           locale: request.query.locale,
         });
         const contract = metadataBundle?.smartContract;
+        const decimals = metadataBundle?.token?.decimals ?? undefined;
+        const total_supply = metadataBundle?.token?.total_supply ?? undefined;
         await reply.send({
           name: metadataBundle?.token?.name ?? undefined,
           symbol: metadataBundle?.token?.symbol ?? undefined,
-          decimals: metadataBundle?.token?.decimals ?? undefined,
-          total_supply: metadataBundle?.token?.total_supply ?? undefined,
+          decimals,
+          total_supply:
+            decimals != undefined && total_supply != undefined
+              ? BigNumber(total_supply)
+                  .dividedBy(10 ** decimals)
+                  .toFixed(decimals)
```

Comment:
I think this could just be something like 
```ts
BigNumber(total_supply).shiftedBy(-decimals)
```
https://mikemcl.github.io/bignumber.js/#shift

Reply:
Cool, thanks, done